solutions/usaco-790.mdx

```diff
@@ -2,14 +2,90 @@
 id: usaco-790
 source: USACO Gold 2018 January
 title: Cow At Large
-author: Qi Wang
+author: Qi Wang, Jesse Choe
 ---
 
 [Official Analysis](http://www.usaco.org/current/data/sol_atlarge_gold_jan18.html)
 
 **Time Complexity:** $\mathcal{O}(N)$
 
+Suppose that a given farmer reaches Bessie at barn $i$; this is only possible if that farmer reached barn $i$ either: 
+
+* At the same time as Bessie
+* At an earlier time than Bessie
+
+This means that we can treat this problem as a shortest path problem, where the farmer reaches barn $i$ at the same time or earlier than Bessie if $\texttt{dist1[i]} \leq \texttt{dist2[i]}$. Here, $\texttt{dist1[i]}$ represents the shortest distance between any given farmer and the $i$-th barn, and $\texttt{dist2[i]}$ represents the shortest distance between Bessie and the $i$-th barn.
+
+We can run BFS to find $\texttt{dist1[i]}$ and $\texttt{dist2[i]}$. Once we find these shortest distances, we can simulate Bessies' movement by "catching" her when $\texttt{dist1[i]} \leq \texttt{dist2[i]}$. Once we catch Bessie, we should increase the total number of farmers by $1$.
```

Comment:
elaborate on the "catching part" maybe
why do we inc the farmer amt by 1 once we catch her? what do we do after?